framework/src/postprocessors/ElementAverageFunctionDifference.C

```diff
@@ -0,0 +1,50 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElementAverageFunctionDifference.h"
+
+#include "Function.h"
+
+registerMooseObject("MooseApp", ElementAverageFunctionDifference);
+
+defineLegacyParams(ElementAverageFunctionDifference);
+
+InputParameters
+ElementAverageFunctionDifference::validParams()
+{
+  InputParameters params = ElementAverageValue::validParams();
+  params.addClassDescription("Calculates the difference between a variable and a function.");
+  params.addRequiredParam<FunctionName>("function", "The analytic solution to compare against");
+  params.addParam<bool>("absolute_value",
+                        false,
+                        "Flag to take the absolute value of the diffference between the variable "
+                        "and the function.");
+  return params;
+}
+
+ElementAverageFunctionDifference::ElementAverageFunctionDifference(
+    const InputParameters & parameters)
+  : ElementAverageValue(parameters),
+    _func(getFunction("function")),
+    _abs(getParam<bool>("absolute_value"))
+{
+}
+
+Real
+ElementAverageFunctionDifference::getValue()
+{
+  const Real integral = ElementAverageValue::getValue();
+
+  gatherSum(_volume);
```

Comment:
I believe you're grabbing the sum of `_volume` twice here as `ElementAverageValue::getValue()` gathers the same sum.

Reply:
Actually... line 42 here is just the average value. Think this should all just be

```
const Real val = ElementAverageValue::getValue();
const Real diff = avg - _func.value(_t, _q_point[_qp])
```

...right?

Reply:
yeah, totally! good catch.